src/main/kotlin/no/skatteetaten/aurora/boober/service/GitService.kt

```diff
@@ -72,22 +72,20 @@ open class GitService(
     @JvmOverloads
     fun checkoutRepository(repositoryName: String, checkoutFolder: String = repositoryName, deleteUnpushedCommits: Boolean = true): Git {
         val repoPath = File(File("$checkoutPath/$checkoutFolder").absoluteFile.absolutePath)
-        val git = if (repoPath.exists()) {
-            try {
-                updateRepository(repoPath, deleteUnpushedCommits)
-            } catch (e: Exception) {
-                logger.warn("Deleting local repository because of update failure. Cause: ${e.message}.")
-                repoPath.deleteRecursively()
-                null
-            }
-        } else null
-
-        if (git != null) return git
+        if (!repoPath.exists()) {
```

Comment:
Jeg må innrømme jeg ikke er døds-fan av denne implementasjonen;
- Den første if-en er egentlig ikke en guard-clause siden den gjør en return fra et metodekall og dermed innfører multiple returnpunkter
- cloneRepository-kallet er duplisert
- Det gjøres litt for mye logikk i catch-blokken (Exception driven logic)

Jeg snakket litt med Espen og foreslår følgende implementasjon:
`    @JvmOverloads
    fun checkoutRepository(repositoryName: String, checkoutFolder: String = repositoryName, deleteUnpushedCommits: Boolean = true): Git {
        val repoPath = File(File("$checkoutPath/$checkoutFolder").absoluteFile.absolutePath)
        val git: Git? = repoPath.takeIf(File::exists).let {
            try {
                updateRepository(repoPath, deleteUnpushedCommits)
            } catch (e: Exception) {
                logger.warn("Deleting local repository because of update failure. Cause: ${e.message}.")
                repoPath.deleteRecursively()
                null
            }
        }

        return git ?: checkoutRepository(repositoryName, repoPath)
    }
`

Reply:
Blir vel cloneRepository istedet for checkoutRepository på slutten der

Reply:
Denne implementasjonen er helt greit for meg, med unntaket av at metoden på bunden ikke bør hete det samme som det denne metoden her gjør. 

Når jeg leste koden første gang trodde jeg vi hadde uendelig rekursjon her siden vi kalte opp en metode med samme navn som oss selv. Siden man har default verdi på parametre i kotlin er jeg ikke veldig fan av å kun bruke parametre for å skille 2 logiske operasjoner fra hveradnre kun med forskjellige parametre og samme navn.

Kan du endre koden og pushe @bentsolheim